File: lib/domain/models/animal_count.dart
import 'package:meta/meta.dart';

@immutable
class AnimalCount {
  final int categoryId;
  final int count;
  const AnimalCount({required this.categoryId, required this.count});

  @override
  bool operator ==(Object other) =>
      other is AnimalCount &&
      other.categoryId == categoryId &&
      other.count == count;

  @override
  int get hashCode => Object.hash(categoryId, count);
}

File: src/domain/models/animal-count.ts
import { z } from 'zod';

export const AnimalCount = z.object({
  categoryId: z.string(),
  count: z.number().int().nonnegative(),
});

export type AnimalCount = z.infer<typeof AnimalCount>;

export const AnimalCountList = z.array(AnimalCount);
export type AnimalCountList = z.infer<typeof AnimalCountList>;
